index.js

```diff
@@ -42,57 +42,57 @@ have been logged anyway.
 */
 
 const debug = require('debug')
+const stringify = require('safe-json-stringify')
 
 // This is the directory from which node process was launched.
 // We use it as root since it gives relative paths which can be clicked on
 // (in terminals such as iTerm) to open the file in an editor.
 const cwd = process.cwd()
 
-expandThunks = (array) => array.map(
+const expandThunks = (array) => array.map(
   elem => typeof elem == 'function' ? elem() : elem
 )
 
-makeLazyLogger = (strictLogger) => {
+const makeLazyLogger = (strictLogger) => {
   return strictLogger.enabled ?
     // if logger enabled, call it with functions expanded to actual values
     (...args) => strictLogger.apply(strictLogger, expandThunks(args)) :
     // no-op
     () => {}
 }
 
-makeStrictLogger = (label, {logToStderr, isErrorLogger}) => {
+const parseArgs = (args) => args
+  .map(arg => stringify(
+    arg instanceof Error ? {
+      name: arg.name,
```

Comment:
could an `Error` object have more properties than these 3 that were added as context?

Reply:
Actually ignore me there, it seems that is not even a valid use case of `Error`, and we have some of that in our codebase. An extension of `Error` could have more properties however so may want to add a `context` field there that we can populate in our custom errors such as `DVFError`

Reply:
There are many error classes in libraries that include extra props (like MongoError) which might be useful to long.

Reply:
Btw. I'm not sure about simply concatenating all args into a string is sticking it all into message. I think we should be able to pass in objects and have those logged in structured way.